core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -132,15 +132,31 @@ class OffsetIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writabl
     }
   }
 
+  /**
+    * Convert given offset to base-relative form. This function throws InvalidOffsetException if the conversion results
+    * in overflow or underflow.
+    * @param offset Offset to convert
+    * @return Base-relative offset
+    * @throws InvalidOffsetException
+    */
+  def relativeOffset(offset: Long): Int = {
```

Comment:
Would it be better to put this method in ` AbstractIndex` given it exists in both `TimeIndex` and `OffsetIndex`?